### PR TITLE
Migrate generated tests for java resttemplate generator updateFeature/resttemplate junit5

### PR DESCRIPTION
**Pull request by timo-a**
Migrates the tests of the generated java client with library `resttemplate` to JUnit5 as discussed here: https://github.com/OpenAPITools/openapi-generator/issues/16352#issuecomment-1933686355
Updating the samples was a bit difficult as existing java files were not overridden. In the end I deleted the relevant sample folders. Some of the changes suggest that the `sample`s hadn't been updated in a while.

Technical commitee menbers are: @bbdouglas, @sreeshas, @jfiala, @lukoyanov, @cbornet, @jeff9finger, @karismann, @Zomzog, @lwlee2608 and @martin-mfg

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
thanks for the PR. can you please run the command to update the samples one more time?

https://github.com/OpenAPITools/openapi-generator/actions/runs/8413416642/job/23035695804?pr=18222

File: modules/openapi-generator/src/main/resources/Java/build.gradle.mustache [hunk 1]
```diff
@@ -125,7 +125,7 @@ ext {
     {{/useBeanValidation}}
     jersey_version = "1.19.4"
     jodatime_version = "2.9.9"
-    junit_version = "4.13.2"
+    junit_version = "5.10.2"
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/Java/build.gradle.mustache:128]: can you please update pom.xml template as well?

Comment (timo-a) [reply modules/openapi-generator/src/main/resources/Java/build.gradle.mustache:128]: I think I've done that already: https://github.com/OpenAPITools/openapi-generator/pull/18222/files/5d06e69dbe323be0cfd3c3df5c68b13b0ebd05fb#diff-545248c48d06e93aa2fe174a53a80ed654eb113f124f7f2f391ed60ea7bb2e91L365-R380

**Comment by timo-a** [comment]
> thanks for the PR. can you please run the command to update the samples one more time?
> 
> https://github.com/OpenAPITools/openapi-generator/actions/runs/8413416642/job/23035695804?pr=18222

done!

**Pull request merged**